cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -269,7 +274,10 @@ public synchronized void updateProperties(Id.Namespace namespaceId, NamespaceMet
     if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
       builder.setSchedulerQueueName(config.getSchedulerQueueName());
     }
-
+    if (config != null && !Strings.isNullOrEmpty(config.getRootDirectory())) {
+      throw new IllegalArgumentException(String.format("%s is immutable and cannot be updated.",
```

Comment:
let's not use immutable, since that is a very software engineering term. Just "cannot be updated" is fine. Also, instead of ``IllegalArgumentException`` here, you should throw ``BadRequestException`` Are there tests for this?